File: AdvantageFramework.Core/Security/Database/Procedures/ClientContact.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Linq;

namespace AdvantageFramework.Core.Security.Database.Procedures
{
    public static partial class ClientContact
    {
        #region Constants

        #endregion

        #region Enums

        #endregion

        #region Variables

        #endregion

        #region Properties

        #endregion

        #region Methods
        public static IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadAllAvailableByClientCode(Security.Database.DbContext DbContext, string ClientCode)
        {
            IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadAllAvailableByClientCodeRet = default;
            LoadAllAvailableByClientCodeRet = from ClientContact in DbContext.ClientContact.AsQueryable()
                                              where ClientContact.ClCode == ClientCode && ClientContact.CpUser == 1 && (ClientContact.InactiveFlag == null || ClientContact.InactiveFlag == 0) && ClientContact.CpUsers.Count == 0
                                              select ClientContact;
            return LoadAllAvailableByClientCodeRet;
        }

        public static IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadByClientCode(Security.Database.DbContext DbContext, string ClientCode)
        {
            IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadByClientCodeRet = default;
            LoadByClientCodeRet = from ClientContact in DbContext.ClientContact.AsQueryable()
                                  where ClientContact.ClCode == ClientCode
                                  select ClientContact;
            return LoadByClientCodeRet;
        }

        public static AdvantageFramework.Core.Security.Database.Entities.ClientContact LoadByClientContactID(Database.DbContext DbContext, int ClientContactID)
        {
            AdvantageFramework.Core.Security.Database.Entities.ClientContact LoadByClientContactIDRet = default;
            try
            {
                LoadByClientContactIDRet = (from ClientContact in DbContext.ClientContact.AsQueryable()
                                            where ClientContact.CdpContactId == ClientContactID
                                            select ClientContact).SingleOrDefault();
            }
            catch (Exception ex)
            {
                LoadByClientContactIDRet = default;
            }

            return LoadByClientContactIDRet;
        }

        public static IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadAllActive(Database.DbContext DbContext)
        {
            IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadAllActiveRet = default;
            LoadAllActiveRet = from ClientContact in DbContext.ClientContact.AsQueryable()
                               where ClientContact.InactiveFlag == null || ClientContact.InactiveFlag == 0
                               select ClientContact;
            return LoadAllActiveRet;
        }

        public static IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> Load(Database.DbContext DbContext)
        {
            IQueryable<AdvantageFramework.Core.Security.Database.Entities.ClientContact> LoadRet = default;
            LoadRet = from ClientContact in DbContext.ClientContact.AsQueryable()
                      select ClientContact;
            return LoadRet;
        }

        #endregion
    }
}

File: AdvantageFramework/Security/Database/Procedures/ClientContact.vb
Namespace Security.Database.Procedures.ClientContact

    <HideModuleName()> _
    Public Module Methods

#Region " Constants "



#End Region

#Region " Enum "



#End Region

#Region " Variables "



#End Region

#Region " Properties "



#End Region

#Region " Methods "

        Public Function LoadAllAvailableByClientCode(ByVal DbContext As Database.DbContext, ByVal ClientCode As String) As System.Data.Entity.Infrastructure.DbQuery(Of Security.Database.Entities.ClientContact)

            LoadAllAvailableByClientCode = From ClientContact In DbContext.GetQuery(Of Database.Entities.ClientContact)
                                           Where ClientContact.ClientCode = ClientCode AndAlso
                                                 ClientContact.IsClientPortalUser = 1 AndAlso
                                                 (ClientContact.IsInactive Is Nothing OrElse
                                                  ClientContact.IsInactive = 0) AndAlso
                                                 ClientContact.ClientPortalUsers.Count = 0
                                           Select ClientContact

        End Function
        Public Function LoadByClientCode(ByVal DbContext As Database.DbContext, ByVal ClientCode As String) As System.Data.Entity.Infrastructure.DbQuery(Of Security.Database.Entities.ClientContact)

            LoadByClientCode = From ClientContact In DbContext.GetQuery(Of Database.Entities.ClientContact)
                               Where ClientContact.ClientCode = ClientCode
                               Select ClientContact

        End Function
        Public Function LoadByClientContactID(ByVal DbContext As Database.DbContext, ByVal ClientContactID As Integer) As Security.Database.Entities.ClientContact

            Try

                LoadByClientContactID = (From ClientContact In DbContext.GetQuery(Of Database.Entities.ClientContact)
                                         Where ClientContact.ContactID = ClientContactID
                                         Select ClientContact).SingleOrDefault

            Catch ex As Exception
                LoadByClientContactID = Nothing
            End Try

        End Function
        Public Function LoadAllActive(ByVal DbContext As Database.DbContext) As System.Data.Entity.Infrastructure.DbQuery(Of Security.Database.Entities.ClientContact)

            LoadAllActive = From ClientContact In DbContext.GetQuery(Of Database.Entities.ClientContact)
                            Where ClientContact.IsInactive Is Nothing OrElse
                                  ClientContact.IsInactive = 0
                            Select ClientContact

        End Function
        Public Function Load(ByVal DbContext As Database.DbContext) As System.Data.Entity.Infrastructure.DbQuery(Of Security.Database.Entities.ClientContact)

            Load = From ClientContact In DbContext.GetQuery(Of Database.Entities.ClientContact)
                   Select ClientContact

        End Function

#End Region

    End Module

End Namespace
